packages/esm-commons-lib/src/hooks/useLastEncounter.tsx

```diff
@@ -0,0 +1,21 @@
+import { openmrsFetch } from '@openmrs/esm-framework';
+import { OpenmrsEncounter } from '../api/types';
+import { encounterRepresentation } from '../constants';
+import useSWR from 'swr';
+
+export function useLastEncounter(patientUuid: string, encounterType: string) {
+  const query = `encounterType=${encounterType}&patient=${patientUuid}&limit=1&order=desc&startIndex=0`;
+    const endpointUrl = `/ws/rest/v1/encounter?${query}&v=${encounterRepresentation}`;
+
+  const { data, error, isValidating } = useSWR<{ data: { results: Array<OpenmrsEncounter> } }, Error>(
+    endpointUrl,
+    openmrsFetch,
+  );
+
+  return {
+    lastEncounter: data ? data?.data?.results.shift() : null,
```

Comment:
@CynthiaKamau are you sure the first element of the array returned is the latest encounter? Can I assume that you taken care of that on the query?